config/tests/jobs/jobs_test.go

```diff
@@ -1086,28 +1086,115 @@ func TestSigReleaseMasterBlockingOrInformingJobsShouldUseFastBuilds(t *testing.T
+// - jobs using --extract=ci/latest-fast MUST pull from gs://k8s-release-dev
+// - release-blocking jobs using --extract=ci/*  MUST from pull gs://k8s-release-dev
+// TODO(https://github.com/kubernetes/k8s.io/issues/846): switch from SHOULD to MUST once all jobs migrated
+// - jobs using --extract=ci/* SHOULD pull from gs://k8s-release-dev
+// TODO(https://github.com/kubernetes/k8s.io/issues/1569): start warning once gs://k8s-release populated
+// - jobs using --extract=release/* SHOULD pull from gs://k8s-release
 func TestKubernetesE2eJobsMustExtractFromK8sInfraBuckets(t *testing.T) {
 	jobs := allStaticJobs()
+	var totalJobs, needsFixJobs int
 	for _, job := range jobs {
-		extract := ""
-		extractCIBucket := "kubernetes-release-dev"
-		for _, arg := range job.Spec.Containers[0].Args {
-			if strings.HasPrefix(arg, "--extract=") {
-				extract = strings.TrimPrefix(arg, "--extract=")
+		needsFix := false
+		extracts := []string{}
+		const (
+			defaultCIBucket       = "kubernetes-release-dev" // ensure this matches kubetest --extract-ci-bucket default
+			expectedCIBucket      = "k8s-release-dev"
+			defaultReleaseBucket  = "kubernetes-release" // ensure this matches kubetest --extract-release-bucket default
+			expectedReleaseBucket = "k8s-release"
+			k8sReleaseIsPopulated = false // TODO(kubernetes/k8s.io#1569): drop this once gs://k8s-release populated
+		)
+		ciBucket := defaultCIBucket
+		releaseBucket := defaultReleaseBucket
+		for _, container := range job.Spec.Containers {
+			for _, arg := range container.Args {
+				if strings.HasPrefix(arg, "--extract=") {
+					extracts = append(extracts, strings.TrimPrefix(arg, "--extract="))
+				}
+				if strings.HasPrefix(arg, "--extract-ci-bucket=") {
+					ciBucket = strings.TrimPrefix(arg, "--extract-ci-bucket=")
+				}
+				if strings.HasPrefix(arg, "--extract-release-bucket=") {
+					releaseBucket = strings.TrimPrefix(arg, "--extract-release-bucket=")
+				}
 			}
-			if strings.HasPrefix(arg, "--extract-ci-bucket=") {
-				extractCIBucket = strings.TrimPrefix(arg, "--extract-ci-bucket=")
+			for _, extract := range extracts {
+				fail := false
+				if extractUsesCIBucket(extract) && ciBucket != expectedCIBucket {
+					needsFix = true
+					jobDesc := "jobs"
+					fail = extract == "ci/latest-fast"
+					if isKubernetesReleaseBlocking(job) {
+						fail = true
+						jobDesc = "release-blocking jobs"
+					}
+					msg := fmt.Sprintf("%s: %s using --extract=%s must have --extract-ci-bucket=%s", job.Name, jobDesc, extract, expectedCIBucket)
+					if fail {
+						t.Errorf("FAIL - %s", msg)
+					} else {
+						t.Logf("WARN - %s", msg)
+					}
+				}
+				if k8sReleaseIsPopulated && extractUsesReleaseBucket(extract) && releaseBucket != expectedReleaseBucket {
+					needsFix = true
+					jobDesc := "jobs"
+					if isKubernetesReleaseBlocking(job) {
+						fail = true
+						jobDesc = "release-blocking jobs"
+					}
+					fail := isKubernetesReleaseBlocking(job)
```

Comment:
this can be dropped?
but anyway this code path isn't exercised yet so can be a follow up

Reply:
doh, thanks, will followup